### PR TITLE
Combobox: Fix cursor jumping to the end of input

### LINKED ISSUE #18499: Cursor jumps to the end on selecting and editing text on freeform combobox
### Environment Information

- **Package version(s)**: 8.10.0, Also tried it on 8.15.0 and still repro.
- **Browser and OS versions**: (fill this out if relevant), not relevant, repro on Edge, Chrome and other browsers.

### Please provide a reproduction of the bug in a codepen:

https://codepen.io/syedinaqvi/pen/abJaOyO

#### Actual behavior:

Go to codepen link here https://codepen.io/syedinaqvi/pen/abJaOyO. Select one option from the dropdown, e.g., select 11:00 PM (ensure that Allow freeform is enabled). Select the 00 part of time and enter 2 numbers to replace the selected value. After first number is pressed cursor jumps to the end.

#### Expected behavior:

Cursor should not jump to the end.

### Priorities and help requested:

Are you willing to submit a PR to fix? No

Requested priority: (Blocking, High, Normal, Low), High, as it repro only on v8.

Products/sites affected: Outlook Web

### PR DESCRIPTION
**Pull request by micahgodbolt**
fixes #18499

In the past, changes to the Autofill component updated internal state and then called the onChange callback. This callback then calculated the new value and updated the value on via getDerivedStateFromProps. This meant that local state updated, component re-renders, but since value is still controlled (and hasn't changed yet), the cursor jumps to the end of the text. After that re-render the value DOES update, but not until after the cursor already jumped.

This change makes it so internal state is not updated if updateValueInWillReceiveProps is defined.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/bc7e2d2fea8c193acb3228b4aff6497ec3ceb029'>bc7e2d2fea8c193acb3228b4aff6497ec3ceb029</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-894d0bed-f8c8-4b12-a243-6a8d22912afe.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-894d0bed-f8c8-4b12-a243-6a8d22912afe.json:1]: ## [🕵 fluentuiv8 Open the Visual Regressions report to inspect the 4 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-fluent/index.html?organization=uifabric&project=fabricpublic&prId=26931&commitId=b1ac67426c5b8fcb3911583bbc03e4285a16fca6&env=prod)

✅ There was 0 screenshots added, 4 screenshots removed, 1041 screenshots unchanged, 0 screenshots with different dimensions and 0 screenshots with visible difference.


  <details><summary><b>unknown</b> 4 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Pivot - Overflow.Root.Narrow - Last tab selected.chromium.png|0| Removed|
|Pivot - Overflow.Root.Narrow - Overflow menu.chromium.png|0| Removed|
|Pivot - Overflow.Tabs - RTL.Narrow - Last tab selected.chromium.png|0| Removed|
|Pivot - Overflow.Tabs - RTL.Narrow - Overflow menu.chromium.png|0| Removed|

</details>


<div id="vrtCommentfluentuiv8"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/BaseButton-mount_base.html">1169</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/BaseButton-mount.html">1179</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Breadcrumb-mount_base.html">2977</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Breadcrumb-mount.html">2914</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Checkbox-mount_base.html">2588</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Checkbox-mount.html">2599</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/CheckboxBase-mount_base.html">2347</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/CheckboxBase-mount.html">2345</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/ChoiceGroup-mount_base.html">4383</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/ChoiceGroup-mount.html">4424</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/ComboBox-mount_base.html">1246</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/ComboBox-mount.html">1264</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/CommandBar-mount_base.html">9535</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/CommandBar-mount.html">9558</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/ContextualMenu-mount_base.html">13829</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/ContextualMenu-mount.html">13897</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/DefaultButton-mount_base.html">1376</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/DefaultButton-mount.html">1386</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/DetailsRow-mount_base.html">3448</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/DetailsRow-mount.html">3493</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/DetailsRowFast-mount_base.html">3470</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/DetailsRowFast-mount.html">3464</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/DetailsRowNoStyles-mount_base.html">3288</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/DetailsRowNoStyles-mount.html">3335</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Dialog-mount_base.html">3064</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Dialog-mount.html">3054</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/DocumentCardTitle-mount_base.html">582</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/DocumentCardTitle-mount.html">576</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Dropdown-mount_base.html">3194</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Dropdown-mount.html">3201</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/FocusTrapZone-mount_base.html">1988</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/FocusTrapZone-mount.html">1989</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/FocusZone-mount_base.html">1958</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/FocusZone-mount.html">1940</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/GroupedList-mount_base.html">51209</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/GroupedList-mount.html">57668</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/GroupedList-virtual-rerender_base.html">23582</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/GroupedList-virtual-rerender.html">23703</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/GroupedList-virtual-rerender-with-unmount_base.html">92517</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/GroupedList-virtual-rerender-with-unmount.html">92101</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/GroupedListV2-mount_base.html">556</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/GroupedListV2-mount.html">557</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/GroupedListV2-virtual-rerender_base.html">548</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/GroupedListV2-virtual-rerender.html">546</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/GroupedListV2-virtual-rerender-with-unmount_base.html">546</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/GroupedListV2-virtual-rerender-with-unmount.html">562</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/IconButton-mount_base.html">1883</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/IconButton-mount.html">1891</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Label-mount_base.html">710</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Label-mount.html">722</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Layer-mount_base.html">4277</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Layer-mount.html">4301</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Link-mount_base.html">819</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Link-mount.html">828</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/MenuButton-mount_base.html">1675</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/MenuButton-mount.html">1664</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/MessageBar-mount_base.html">2321</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/MessageBar-mount.html">2314</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Nav-mount_base.html">3261</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Nav-mount.html">3240</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/OverflowSet-mount_base.html">1359</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/OverflowSet-mount.html">1368</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Panel-mount_base.html">2598</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Panel-mount.html">2514</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Persona-mount_base.html">1318</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Persona-mount.html">1315</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Pivot-mount_base.html">1642</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Pivot-mount.html">1641</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/PrimaryButton-mount_base.html">1521</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/PrimaryButton-mount.html">1522</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Rating-mount_base.html">6914</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Rating-mount.html">6966</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/SearchBox-mount_base.html">1512</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/SearchBox-mount.html">1503</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Shimmer-mount_base.html">2866</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Shimmer-mount.html">2838</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Slider-mount_base.html">2074</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Slider-mount.html">2061</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/SpinButton-mount_base.html">4635</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/SpinButton-mount.html">4650</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Spinner-mount_base.html">804</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Spinner-mount.html">812</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/SplitButton-mount_base.html">3060</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/SplitButton-mount.html">3073</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Stack-mount_base.html">817</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Stack-mount.html">825</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/StackWithIntrinsicChildren-mount_base.html">2325</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/StackWithIntrinsicChildren-mount.html">2334</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/StackWithTextChildren-mount_base.html">4746</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/StackWithTextChildren-mount.html">4752</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/SwatchColorPicker-mount_base.html">10121</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/SwatchColorPicker-mount.html">10428</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/TagPicker-mount_base.html">2604</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/TagPicker-mount.html">2588</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/TeachingBubble-mount_base.html">83963</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/TeachingBubble-mount.html">84416</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Text-mount_base.html">798</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Text-mount.html">772</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/TextField-mount_base.html">1560</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/TextField-mount.html">1583</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/ThemeProvider-mount_base.html">1528</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/ThemeProvider-mount.html">1523</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/ThemeProvider-virtual-rerender_base.html">1077</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/ThemeProvider-virtual-rerender.html">1085</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">2154</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">2164</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Toggle-mount_base.html">1095</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/Toggle-mount.html">1093</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/buttonNative-mount_base.html">539</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26931/perf-test/buttonNative-mount.html">552</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: packages/react/src/components/Autofill/Autofill.tsx [hunk 2]
```diff
@@ -326,12 +326,16 @@ export class Autofill extends React.Component<IAutofillProps, IAutofillState> im
...
 
     // eslint-disable-next-line deprecation/deprecation
-    const { onInputChange, onInputValueChange } = this.props;
+    const { onInputChange, onInputValueChange, updateValueInWillReceiveProps } = this.props;
     if (onInputChange) {
       newValue = onInputChange?.(newValue, composing) || '';
     }
 
-    this.setState({ inputValue: newValue }, () => onInputValueChange?.(newValue, composing));
+    // if value is controlled, then we should not update the value in state
```
Comment (smhigley) [inline packages/react/src/components/Autofill/Autofill.tsx:334]: minor nit: this is only when controlled through `updateValueInWillReceiveProps`, and not when controlled through `props.value`

```suggestion
    // if value is controlled in updateValueInWillReceiveProps, then we should not update the value in state now
```

**Review by smhigley — APPROVED** [review]

**Pull request merged**